ML_Project_1.ipynb

```diff
@@ -0,0 +1 @@
2DwAfTuN7FDilimtiTc6zX5hZ4blGZmaF50RmZoXnRGZmhedEZmaF50RmZoXnRGZmhedEZmaF9/8BfhsuhEaPNfYAAAAASUVORK5CYII=\n"},"metadata":{"needs_background":"light"}}]},{"metadata":{},"cell_type":"markdown","source":["Part 5 (continued):\n","\n","2. It seems that support vector machines perform the worst, with K-nearest neighbours coming in second worst. Looking at their performance on the validation data however, it seems that K-nearest neighbours achieved a relatively high accuracy of above 80%, which means that this model was perhaps overfitting to the training data and failing to generalise to new examples. Support vector machines did poorly on the validation data and much worse on the test data. SVMs work best with data that has clearly defined margins of separation between groups, meaning that it might not have been clear with this data. This is best shown with the visualisation of the data above, and seeing that there are no distinct boundaries between the three classes. Logistic Regression and Random Forests seem to be performing the best on the data, sugessting that the data is more suited for them.\n","\n","3. The best permutation found was Logistic Regression with an accuracy of 95.29% and a C parameter value of 1.4062 using the L2 norm. Since larger values of C indicate weaker regularistion, a value slightly larger than 1 suggests slighty weak use of regularistion was the best. This model generalises very well, as the accuracy only dropped a little on the test data from the validation data, whereas other models like Random Forests had a perfect accuracy on the validation data but an ever larger drop in accuracy on test data. This suggests there is a good balance between variance and bias with this Logistic Regression model. The confusion matrices above also show that there was a very low number of false positives and false negatives with this model, suggesting a high precision and recall as well as accuracy."]},{"metadata":{},"cell_type":"markdown","source":["## Part 6: \"A stakeholder asks you which features most affect the response variable (output). Describe how you would organise a test to determine this.\"\n","\n","I would test this by manipulating the input data for the models, such that all the Xs for one feature are set to zero, and then repeating this until every feature has had a chance to be set to zero. I would then compare which result ends up with the biggest difference from the original result which had all features included. The feature when set to zero that correspends to the biggest difference in accuracy, would therefore be the feature that has the greatest influence over the response variable. "]}],"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"name":"python","version":"3.8.6-final","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat":4,"nbformat_minor":2}
```

Comment:
<p>pass in the model class rather than the model name</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AI-Core-Assignments/project-1-SherZach/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>